addons/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/api/FreeboxApiManager.java

```diff
@@ -0,0 +1,498 @@
+        try {
+            Properties headers = null;
+            String token = getSessionToken();
+            if (token != null) {
+                headers = new Properties();
+                headers.setProperty(AUTH_HEADER, token);
+            }
+            InputStream stream = null;
+            String contentType = null;
+            if (requestContent != null) {
+                stream = new ByteArrayInputStream(requestContent.getBytes(StandardCharsets.UTF_8));
+                contentType = HTTP_CALL_CONTENT_TYPE;
+            }
+            logger.debug("executeUrl {} {} requestContent {}", httpMethod, relativeUrl,
+                    doNotLogData ? "***" : requestContent);
+            String jsonResponse = HttpUtil.executeUrl(httpMethod, baseAddress + relativeUrl, headers, stream,
+                    contentType, HTTP_CALL_DEFAULT_TIMEOUT_MS);
+            if (stream != null) {
+                stream.close();
+                stream = null;
+            }
+
+            if (patchTableReponse) {
+                // Replace empty result by an empty table result
+                jsonResponse = jsonResponse.replace("\"result\":{}", "\"result\":[]");
+            }
+
+            return evaluateJsonReesponse(jsonResponse, responseClass, doNotLogData);
+        } catch (FreeboxException e) {
+            if (retryAuth && e.isAuthRequired()) {
+                logger.debug("Authentication required: open a new session and retry the request");
+                openSession();
+                return executeUrl(httpMethod, relativeUrl, requestContent, responseClass, false, patchTableReponse,
+                        doNotLogData);
+            }
+            throw e;
+        } catch (IOException e) {
+            throw new FreeboxException(httpMethod + " request " + relativeUrl + ": execution failed: " + e.getMessage(),
+                    e);
+        } catch (JsonSyntaxException e) {
+            throw new FreeboxException(
+                    httpMethod + " request " + relativeUrl + ": response parsing failed: " + e.getMessage(), e);
+        }
+    }
+
+    private <T extends FreeboxResponse<F>, F> F evaluateJsonReesponse(String jsonResponse, Class<T> responseClass,
+            boolean doNotLogData) throws JsonSyntaxException, FreeboxException {
+        logger.debug("evaluateJsonReesponse Json {}", doNotLogData ? "***" : jsonResponse);
+        // First check only if the result is successful
+        FreeboxResponse<Object> partialResponse = gson.fromJson(jsonResponse, FreeboxEmptyResponse.class);
+        partialResponse.evaluate();
+        // Parse the full response in case of success
+        T fullResponse = gson.fromJson(jsonResponse, responseClass);
+        fullResponse.evaluate();
+        F result = fullResponse.getResult();
+        return result;
+    }
+
+    private String encodeUrl(String url) throws FreeboxException {
+        try {
+            return URLEncoder.encode(url, "UTF-8");
```

Comment:
Minor: you could use `StandardCharsets.UTF_8` or `StandardCharsets.UTF_8.name()` here

Reply:
Done